### PR TITLE
ntfy 2.3.1 (new formula)

### PR DESCRIPTION
**Pull request by Moulick**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----
Continuation of https://github.com/Homebrew/homebrew-core/pull/116607

### CODE REVIEW

File: Formula/ntfy.rb
Comment (carlocab) [inline Formula/ntfy.rb:41]:
```suggestion

    ntfy_in = shell_output("#{bin}/ntfy publish #{random_topic} 'Test message from HomeBrew during build'")
    ohai ntfy_in
    sleep 1
    ntfy_out = shell_output("#{bin}/ntfy subscribe --poll #{random_topic}")
    ohai ntfy_out
    assert_match ntfy_in, ntfy_out
```
You don't need the `begin` ... `end` block here.

Comment (Moulick) [reply Formula/ntfy.rb]: Thanks for the pointer, fixed!

File: Formula/ntfy.rb
Comment (carlocab) [inline Formula/ntfy.rb:4]:
```suggestion
  url "https://github.com/binwiederhier/ntfy.git",
      tag: "v2.3.1",
      revision: "a75fb08ef138d67599499c1f78628a0b35fcef54"
```
Let's follow the format used by other formulae. Mind the spacing for `tag:` and `revision:`. They are aligned with the first argument to `url`.

File: Formula/ntfy.rb
Comment (carlocab) [inline Formula/ntfy.rb:11]:
```suggestion
```
I don't think you need this. To see, check the output of `brew livecheck ntfy` with and without this block.

**Review by carlocab — COMMENTED** [review]
Thanks; just a few more minor comments. Apologies for missing them in my first review.

Please also squash all your commits together when you make these changes. This will help you get rid of the https://github.com/Homebrew/homebrew-core/labels/automerge-skip label.

File: Formula/ntfy.rb
Comment (Moulick) [reply Formula/ntfy.rb:4]: Sure! Thanks for pointing this out. Maybe this can also be added as the check, in the brew style check?

Comment (carlocab) [reply Formula/ntfy.rb]: Yes, ideally that would be, but we haven't gotten around to adding it to our style checks.

File: Formula/ntfy.rb
Comment (carlocab) [reply Formula/ntfy.rb:13]: Did you forget the `livecheck` block? I checked myself and it doesn't look like it's needed.

File: Formula/ntfy.rb
Comment (carlocab) [inline Formula/ntfy.rb:6]:
```suggestion
      tag:      "v2.3.1",
      revision: "a75fb08ef138d67599499c1f78628a0b35fcef54"
```
Forgot about this one. This one is part of a style check.

File: Formula/ntfy.rb
Comment (Moulick) [reply Formula/ntfy.rb:13]: I tested now locally without the block, it is able to livecheck even without it

File: Formula/ntfy.rb
Comment (Moulick) [reply Formula/ntfy.rb:6]: yep, fixing

**Comment by Moulick** [comment]
ah I f'ed up the squash 🤦

**Comment by carlocab** [comment]
Looks like you accidentally picked up some extra commits. To get rid of them:
```sh
cd "$(brew --repository homebrew/core)"
git checkout new-ntfy
git fetch origin
git rebase origin/master
```

**Comment by Moulick** [comment]
> Looks like you accidentally picked up some extra commits. To get rid of them:
> 
> ```shell
> cd "$(brew --repository homebrew/core)"
> git checkout new-ntfy
> git fetch origin
> git rebase origin/master
> git push --force
> ```

Yep, fixed now.

**Review by carlocab — APPROVED** [review]
Nice work!

**Comment by Moulick** [comment]
Interesting that even after I force-pushed a commit after your approval @carlocab, GitHub has not removed your approval 🤯.

**Comment by p-linnane** [comment]
@Moulick Thanks for making us aware. We are working to rectify this.

**Review by p-linnane — APPROVED** [review]

**Comment by Moulick** [comment]
closes https://github.com/binwiederhier/ntfy/issues/286

**Comment by binwiederhier** [comment]
Awesome, great work. Added install instructions https://docs.ntfy.sh/install/#homebrew :heart:

**Pull request merged**
